service/project.go

```diff
@@ -592,9 +592,8 @@ func (uis *UIServer) modifyProject(w http.ResponseWriter, r *http.Request) {
 	for i, buildDef := range responseRef.PeriodicBuilds {
 		catcher.Wrapf(buildDef.Validate(), "invalid periodic build definition on line %d", i+1)
 	}
-	for _, containerResource := range responseRef.ContainerSizes {
-		catcher.NewWhen(utility.FromIntPtr(containerResource.CPU) <= 0, "container resource CPU must be a positive integer")
-		catcher.NewWhen(utility.FromIntPtr(containerResource.MemoryMB) <= 0, "container resource Memory MB must be a positive integer")
+	for _, containerResource := range containerSizes {
+		catcher.Add(model.ValidateContainerSize(containerResource))
```

Comment:
For a more user-friendly error message, I would wrap the error to include the container size name that was invalid:
```go
catcher.Wrapf(model.ValidateContainerSize(containerResource), "invalid container size '%s'", size)
```